test/legacy_test/test_histogramdd_op.py

```diff
@@ -0,0 +1,640 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+
+import paddle
+
+
+# inputs, bins, range, weights, density
+class TestHistogramddAPI(unittest.TestCase):
+    def setUp(self):
+        self.range = None
+        self.weights = None
+        self.density = False
+
+        self.init_input()
+        self.set_expect_output()
+        self.place = (
+            paddle.CUDAPlace(0)
+            if paddle.is_compiled_with_cuda()
+            else paddle.CPUPlace()
+        )
+
+    def init_input(self):
+        self.sample = np.array([[0.0, 1.0], [1.0, 0.0], [2.0, 0.0], [2.0, 2.0]])
+        self.bins = [3, 3]
+        self.weights = np.array([1.0, 2.0, 4.0, 8.0])
+
+    def set_expect_output(self):  # pytorch output as expect_hist/expect_edges
+        self.expect_hist = np.array(
```

Comment:
我看你结果都使用的np.array，该结果无法与numpy.histogramdd对齐吗？

Reply:
> 我看你结果都使用的np.array，该结果无法与numpy.histogramdd对齐吗？

因为numpy.histogramdd不支持多维(只支持(N,D)两维)，而torch中支持多维，所以这里用了torch的输出来作为reference

Reply:
好的

Reply:
It is best to use `numpy.historamdd` for testing, rather than this fixed result approach. If `numpy.historamdd` does not support ND array, you can use `x= x.reshape(-1, x.shape[-1])` to convert the input ndarray of `x` to 2D array

Reply:
> It is best to use `numpy.historamdd` for testing, rather than this fixed result approach. If `numpy.historamdd` does not support ND array, you can use `x= x.reshape(-1, x.shape[-1])` to convert the input ndarray of `x` to 2D array

Tks for your review, I think I know what you mean~